pkgs/desktops/cinnamon/muffin/default.nix

```diff
@@ -0,0 +1,102 @@
+{ fetchFromGitHub
+, autoreconfHook
+, cinnamon-desktop
+, glib
+, gnome3
+, gnome-doc-utils
+, fetchpatch
+, gobject-introspection
+, gtk3
+, intltool
+, json-glib
+, libinput
+, libstartup_notification
+, libXtst
+, libxkbcommon
+, pkgconfig
+, stdenv
+, udev
+, xorg
+, wrapGAppsHook
+, pango
+, cairo
+, gtk-doc
+, docbook_xsl
+, docbook_xml_dtd_43
+, docbook_xml_dtd_42
+, docbook_xml_dtd_412
+}:
+
+# it's a frankensteins monster with some cinnamon sparkles added on top of it
+
+stdenv.mkDerivation rec {
+  pname = "muffin";
+  version = "4.4.2";
+
+  src = fetchFromGitHub {
+    owner = "linuxmint";
+    repo = pname;
+    rev = version;
+    sha256 = "1kzjw4a5p69j8x55vpbpn6gy8pkbbyii6kzw2nzbypmipgnnijw8";
+  };
+
+  patches = [
+    (fetchpatch { # https://github.com/linuxmint/muffin/pull/548#issuecomment-578316820
+      url = "https://github.com/linuxmint/muffin/commit/f78bf5b309b3d306848f47cc241b31e9399999a7.patch";
+      sha256 = "1c79aa9w2v23xlz86x3l42pavwrqx5d6nmfd9nms29hjsk8mpf4i";
+    })
+    (fetchpatch { # https://github.com/linuxmint/muffin/issues/535#issuecomment-536917143
+      url = "https://src.fedoraproject.org/rpms/muffin/blob/master/f/0001-fix-warnings-when-compiling.patch";
+      sha256 = "15wdbn3afn3103v7rq1icp8n0vqqwrrya03h0g2rzqlrsc7wrvzw";
+    })
+  ];
+
+  buildInputs = [
+    gtk3
+    glib
+    pango
+    cairo
+    json-glib
+    cinnamon-desktop
+    xorg.libXcomposite
+    xorg.libXcursor
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXi
+    xorg.libxkbfile
+    xorg.xkeyboardconfig
+
+    libxkbcommon
+    gnome3.zenity
+    gnome-doc-utils
+    libinput
+    libstartup_notification
+    libXtst
+    udev
+    gobject-introspection
+  ];
+
+  nativeBuildInputs = [
+    autoreconfHook
+    wrapGAppsHook
+    pkgconfig
+    intltool
+
+    gtk-doc
+    docbook_xsl
+    docbook_xml_dtd_43
+    docbook_xml_dtd_42
+    docbook_xml_dtd_412
+  ];
+
+  preAutoreconf = "NOCONFIGURE=1 ./autogen.sh";
```

Comment:
We probably should not use `autoreconfHook` if we are running `autogen.sh` (it will run autoconf & co. twice). Instead we should add the dependencies explicitly and run this in `preConfigure`.

Reply:
does autoreconf hook add any flags to autotools? otherwise those (like destination for installing or whatever) would be missed (unless that's fully handled by configure phase)

Reply:
`autoreconfHook` just runs `autoreconf`, just like `autogen.sh`:

https://github.com/NixOS/nixpkgs/blob/4ad4523600c293a03ca2d51b1bee61e83170de90/pkgs/build-support/setup-hooks/autoreconf.sh#L5

The flags are indeed passed in `configurePhase`.